lessons/find_number.rb

```diff
@@ -0,0 +1,56 @@
+# method to validate an input number
+def valid_number?(input_number)
+    return false unless input_number.strip.match?(/^\d{1,3}$/)
+
+    number = input_number.to_i
+    (1..100).include?(number)
+end
+
+# method to ply the game + input validation + input and random numbers comparison
+def guessing_game
+    secret_number = rand(1..100)
+
+    loop do
+        puts "Guess a number from 1 through 100"
+        user_guess = gets.chomp
+        unless valid_number?(user_guess)
+            puts "Invalid input. Please enter a number between 1 and 100 inclusive:"
```

Comment:
nit: since you already have a prompt asking for the number on line 14, it is enough to say "Invalid input" here.

Reply:
Oh I see what you mean. I just thought if a user entered invalid number, for ex. less than 0 or more than 100, it would be good to remind them the range of the numbers we need